realm/realm-library/src/main/java/io/realm/RealmCollection.java

```diff
@@ -0,0 +1,162 @@
+     * {@code null} values are ignored.
+     * @throws java.lang.IllegalArgumentException if fieldName is not a Date field.
+     * @throws java.lang.IllegalStateException if the Realm has been closed or called from an incorrect thread.
+     */
+    Date maxDate(String fieldName);
+
+    /**
+     * Finds the minimum date.
+     *
+     * @param fieldName the field to look for the minimum date. If fieldName is not of Date type, an exception is
+     *                  thrown.
+     * @return if no objects exist or they all have {@code null} as the value for the given date field, {@code null}
+     * will be returned. Otherwise the minimum date is returned. When determining the minimum date, objects with
+     * {@code null} values are ignored.
+     * @throws java.lang.IllegalArgumentException if fieldName is not a Date field.
+     * @throws java.lang.IllegalStateException if the Realm has been closed or called from an incorrect thread.
+     */
+    Date minDate(String fieldName);
+
+    /**
+     * This deletes all objects in the collection from the underlying Realm as well as from the collection.
+     *
+     * @throws IllegalStateException if the corresponding Realm is closed or in an incorrect thread.
+     * @return {@code true} if objects was deleted, {@code false} otherwise.
+     * @throws java.lang.IllegalStateException if the Realm has been closed or called from an incorrect thread.
+     */
+    boolean deleteAllFromRealm();
+
+    /**
+     * Checks if a collection has finished loading its data yet.
+     *
+     * @return {@code true} if data has been loaded and is available, {@code false} if data is still being loaded.
+     */
+    boolean isLoaded();
+
+    /**
+     * Blocks the collection until all data are available.
+     *
+     * @return {@code true} if the data could be successfully loaded, {@code false} otherwise.
+     */
+    boolean load();
+
+    /**
+     * Checks if the collection is still valid to use e.g. the {@link io.realm.Realm} instance hasn't
+     * been closed.
+     *
+     * @return {@code true} if still valid to use, {@code false} otherwise.
+     */
+    boolean isValid();
+
+    /**
+     * Returns the Realm instance backing this collection.
+     *
+     * @return the {@link BaseRealm} backing this collection or {@code null} if it is an un-managed collection.
+     */
+    BaseRealm getRealm();
+
+    /**
+     * Tests whether this {@code Collection} contains the specified object. Returns
+     * {@code true} if and only if at least one element {@code elem} in this
+     * {@code Collection} meets following requirement:
+     * {@code (object==null ? elem==null : object.equals(elem))}.
+     *
+     * @param object the object to search for.
+     * @return {@code true} if object is an element of this {@code Collection}, {@code false} otherwise.
+     * @throws NullPointerException if the object to look for is {@code null} and this {@code Collection} doesn't
```

Comment:
This is the standard behaviour in the standard Collection interface, so I would rather keep this.